github/repos_actions_allowed.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// GetActionsAllowed gets the actions that are allowed in a repository.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/actions#get-allowed-actions-and-workflows-for-a-repository
+func (s *RepositoriesService) GetActionsAllowed(ctx context.Context, org string, repo string) (*ActionsAllowed, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/permissions/selected-actions", org, repo)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	actionsAllowed := new(ActionsAllowed)
+	resp, err := s.client.Do(ctx, req, actionsAllowed)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return actionsAllowed, resp, nil
+}
+
+// EditActionsAllowed sets the actions that are allowed in a repository.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/actions#set-allowed-actions-and-workflows-for-a-repository
+func (s *RepositoriesService) EditActionsAllowed(ctx context.Context, org string, repo string, actionsAllowed ActionsAllowed) (*ActionsAllowed, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/permissions/selected-actions", org, repo)
+	req, err := s.client.NewRequest("PUT", u, actionsAllowed)
+	if err != nil {
+		return nil, nil, err
+	}
+	p := new(ActionsAllowed)
+	resp, err := s.client.Do(ctx, req, p)
+	return p, resp, err
```

Comment:
Let's please follow the same pattern here of explicit err and success paths:

```
	if err != nil {
		return nil, resp, err
	}

	return p, resp, nil
```

Reply:
Ah, ok now I finally begin to understand that pattern 👍 

Just a remark: I just copied from the existing [org level code](https://github.com/h2floh/go-github/blob/24a732e7e42d188235682fedf979c3a0ff49e1d6/github/orgs_actions_permissions.go#L57).

Reply:
> Just a remark: I just copied from the existing [org level code](https://github.com/h2floh/go-github/blob/24a732e7e42d188235682fedf979c3a0ff49e1d6/github/orgs_actions_permissions.go#L57).

Thank you, @h2floh !  Please take a look at #2318 when you get a chance and LGTM+Approve if you agree with the changes.